core/app/models/spree/address.rb

```diff
@@ -225,7 +225,7 @@ def should_geocode?
     def set_default_values
       self.firstname ||= user.first_name
       self.lastname ||= user.last_name
-      self.phone ||= user.phone
+      self.phone ||= user.phone if user.respond_to?(:phone)
```

Comment:
@mad-eel please add a migration to add `phone` column to `Spree.user_class.table_name`